gpiozero/output_devices.py

```diff
@@ -300,7 +299,10 @@ def __init__(self, pin=None, active_high=True, initial_value=0, frequency=100):
         self._controller = None
         if not 0 <= initial_value <= 1:
             raise OutputDeviceBadValue("initial_value must be between 0 and 1")
-        super(PWMOutputDevice, self).__init__(pin, active_high)
+        if initial_value in (0, 1):
+            super(PWMOutputDevice, self).__init__(pin, active_high, initial_value=initial_value)
```

Comment:
Okay. So when the Pin classes *do* get fixed to set the frequency-and-value together (e.g. a `pwm_with_state` function, analogous to `output_with_state`), then I guess there'll also need to be some mechanism added to stop the parent OutputDevice class either calling `output_with_state` or setting the `function` property?